packages/react/src/index.tsx

```diff
@@ -1,47 +1,75 @@
-import React, { createContext, useContext, useEffect, useState } from 'react'
+import React, { PropsWithChildren, createContext, useContext, useEffect, useState } from 'react'
 import ButtonImg from './medium-passport-button.svg'
 
-const ParallelContext = createContext({ parallel: null })
+import { ParallelApiRecord, loadParallel } from '@parallelmarkets/vanilla'
+import type { AuthCallbackResult, Parallel } from '@parallelmarkets/vanilla'
+import { AccreditationsApiResponse } from './accreditation_api_types'
+import { ProfileApiResponse } from './profile_api_types'
+import { BlockchainApiResponse } from './blockchain_api_types'
+import { IdentityApiResponse } from './identity_api_types'
 
-export const ParallelProvider = ({ parallel, children, ...props }) => {
+export * from './accreditation_api_types'
+export * from './profile_api_types'
+export * from './blockchain_api_types'
+export * from './identity_api_types'
+
+type LoadParallelPromise = ReturnType<typeof loadParallel>
+type LoadParallelResult = Awaited<ReturnType<typeof loadParallel>>
+type ParallelContextValue = { parallel?: LoadParallelPromise }
+
+const ParallelContext = createContext<ParallelContextValue>({ parallel: undefined })
+
+type ParallelProviderProps = PropsWithChildren<{
+  parallel: LoadParallelPromise
+}>
+
+export const ParallelProvider = ({ parallel, children, ...props }: ParallelProviderProps) => {
   return (
     <ParallelContext.Provider value={{ parallel }} {...props}>
       {children}
     </ParallelContext.Provider>
   )
 }
 
-const isPromise = (thing) => {
-  return thing !== null && typeof thing === 'object' && typeof thing.then === 'function'
+const isPromise = (thing: unknown): thing is PromiseLike<unknown> => {
+  return typeof (thing as PromiseLike<unknown>)?.then === 'function'
 }
 
-const wrapApiCall = (parallel, endpoint) => {
+// The Embed API works with callback functions. This wrapper converts them to promises.
+const promisifyApiCall = <ResultType extends ParallelApiRecord>(parallel: Parallel, endpoint: string) => {
```

Comment:
This function has a Generic Type named `ResultType`. It's constrained to be at least like a `ParallelApiRecord` type via `extends`. 

We use this to return a tightly-typed result from the Promise, down by the `resolve()` with the `as`. This way, we have just 1 function called `promisifyApiCall` but we can use it with all 4 of our API Return Types (we have 4 of those) without having to write out 4 different ApiCall function like this one.